pkg/operator/controller/ingress/status.go

```diff
@@ -105,6 +106,18 @@ func MergeConditions(conditions []operatorv1.OperatorCondition, updates ...opera
 	return conditions
 }
 
+// PruneConditions removes any conditions that are not currently supported.
+// Returns the updated condition array.
+func PruneConditions(conditions []operatorv1.OperatorCondition) []operatorv1.OperatorCondition {
+	for i, condition := range conditions {
+		switch condition.Type {
+		case "DeploymentDegraded":
```

Comment:
We stopped using `DeploymentDegraded` in 4.6, right? Might be worth dropping a comment above indicating that this condition exists for clusters < 4.6, to avoid any confusion.

Reply:
That's a good idea. I'll include a comment about it to clarify